apps/server/src/shared/repo/federalstate/federal-state.repo.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@nestjs/common';
```

Comment:
Repos are not places in shared in future. I do not like to add new repos on this place. 
It is use only in school module atm? Then we can move it to this place. 

https://github.com/hpi-schul-cloud/schulcloud-server/tree/main/apps/server/src/modules/files-storage/repo

For the entity it self it is the same. But harder to solve without dependency cicles in schools. Maybe you can use the import type for it. I am unsure what is part of a fix what not. But this PR include a lot of new code.